File: src/projeto/Validador.java
package projeto;

public class Validador {
    public static boolean validarEndereco(String endereco) {
        String regex = "^BR[a-fA-F0-9]{40}$"; 
        return endereco.matches(regex);
    }

    public static boolean validarTransacao(Transacao transacao) {
        return validarEndereco(transacao.getRemetente()) &&
               validarEndereco(transacao.getDestinatario()) &&
               transacao.getQuantia() > 0;
    }
}

File: src/validador.js
class Validador {
    static validarEndereco(endereco) {
        const regex = /^BAT[a-fA-F0-9]{40}$/;
        return regex.test(endereco);
    }

    static validarTransacao(transacao) {
        return (
            this.validarEndereco(transacao.remetente) &&
            this.validarEndereco(transacao.destinatario) &&
            transacao.quantia > 0 &&
            transacao.taxas >= 0
        );
    }
}

module.exports = Validador;